syntax = "proto2";

// generated with https://github.com/egirault/googleplay-api/blob/master/androguard/androproto.py
// then modified a bit by hand to add the new fields that were added after the move to gms from gsf
// also look at https://github.com/microg/android_packages_apps_GmsCore/blob/master/play-services-core/src/main/protos-repo/checkin.proto
package playapi.proto.gsf;

message AndroidCheckinRequest {
    optional string imei = 1;
    optional int64 id = 2;
    optional string digest = 3;
    optional AndroidCheckinProto checkin = 4;
    optional string desiredBuild = 5;
    optional string locale = 6;
    optional int64 loggingId = 7;
    optional string marketCheckin = 8;
    repeated string macAddr = 9;
    optional string meid = 10;
    repeated string accountCookie = 11;
    optional string timeZone = 12;
    optional fixed64 securityToken = 13;
    optional int32 version = 14;
    repeated string otaCert = 15;
    optional string serialNumber = 16;
    optional string esn = 17;
    optional DeviceConfigurationProto deviceConfiguration = 18;
    repeated string macAddrType = 19;
    optional int32 fragment = 20;
    optional string userName = 21;
    optional int32 userSerialNumber = 22;
    optional string droidguardResult = 24;
    optional string deviceDataVersionInfo = 25;
    optional bool fetchSystemUpdates = 29;
}
message AndroidCheckinResponse {
    optional bool statsOk = 1;
    repeated AndroidIntentProto intent = 2;
    optional int64 timeMsec = 3;
    optional string digest = 4;
    repeated GservicesSetting setting = 5;
    optional bool marketOk = 6;
    optional fixed64 androidId = 7;
    optional fixed64 securityToken = 8;
    optional bool settingsDiff = 9;
    repeated string deleteSetting = 10;
    optional string deviceDataVersionInfo = 12;
}
message GservicesSetting {
    optional bytes name = 1;
    optional bytes value = 2;
}
message DeviceConfigurationProto {
    message FeatureWithGLVersion {
        optional string name = 1;
        optional int32 glEsVersion = 2;
    }
    optional int32 touchScreen = 1;
    optional int32 keyboard = 2;
    optional int32 navigation = 3;
    optional int32 screenLayout = 4;
    optional bool hasHardKeyboard = 5;
    optional bool hasFiveWayNavigation = 6;
    optional int32 screenDensity = 7;
    optional int32 glEsVersion = 8;
    repeated string systemSharedLibrary = 9;
    repeated string systemAvailableFeature = 10;
    repeated string nativePlatform = 11;
    optional int32 screenWidth = 12;
    optional int32 screenHeight = 13;
    repeated string systemSupportedLocale = 14;
    repeated string glExtension = 15;
    optional int32 maxApkDownloadSizeMb = 17;
    optional int32 smallestScreenWidthDp = 18;
    optional bool lowRamDevice = 19;
    optional int64 totalMemoryBytes = 20;
    optional int32 maxNumOfCpuCores = 21;
    repeated FeatureWithGLVersion newSystemAvailableFeature = 26;
    optional int32 screenLayout2 = 27;
    optional bool keyguardDeviceSecure = 28;
}
message AndroidBuildProto {
    message PackageVersion {
        optional int32 version = 1;
        optional string name = 2;
    }
    optional string id = 1;
    optional string product = 2;
    optional string carrier = 3;
    optional string radio = 4;
    optional string bootloader = 5;
    optional string client = 6;
    optional int64 timestamp = 7;
    optional int32 googleServices = 8;
    optional string device = 9;
    optional int32 sdkVersion = 10;
    optional string model = 11;
    optional string manufacturer = 12;
    optional string buildProduct = 13;
    optional bool otaInstalled = 14;
    repeated PackageVersion googlePackage = 15;
    optional string securityPatch = 19;
}
message AndroidCheckinReasonProto {
    optional int32 reasonType = 1; // what are the possible reason types?
    optional int32 attemptCount = 2;
    optional string sourcePackage = 3;
    optional string sourceClass = 4;
    optional bool sourceForce = 5;
}
message AndroidCheckinProto {
    optional AndroidBuildProto build = 1;
    optional int64 lastCheckinMsec = 2;
    repeated AndroidEventProto event = 3;
    repeated AndroidStatisticProto stat = 4;
    repeated string requestedGroup = 5;
    optional string cellOperator = 6;
    optional string simOperator = 7;
    optional string roaming = 8;
    optional int32 userNumber = 9;
    optional int32 deviceType = 14; // 1 - unknown, 2 - phone, 3 - tablet, 4 - tv,  5 - glass, 7 - wearable
    optional AndroidCheckinReasonProto reason = 15;
    optional bool voiceCapable = 18;
}
message AndroidEventProto {
    optional string tag = 1;
    optional string value = 2;
    optional int64 timeMsec = 3;
}
message AndroidIntentProto {
    optional string action = 1;
    optional string dataUri = 2;
    optional string mimeType = 3;
    optional string javaClass = 4;
    repeated group Extra = 5 {
        optional string name = 6;
        optional string value = 7;
    }
}
message AndroidStatisticProto {
    optional string tag = 1;
    optional int32 count = 2;
    optional float sum = 3;
}